scripts/setup-helper-functions.sh

```diff
@@ -132,7 +132,32 @@ function get_cxx_flags {
     ;;
 
     "aarch64")
-      echo -n "-mcpu=neoverse-n1 -std=c++17 $ADDITIONAL_FLAGS"
+      # Read Arm MIDR_EL1 register to detect Arm cpu.
+      # https://developer.arm.com/documentation/100616/0301/register-descriptions/aarch64-system-registers/midr-el1--main-id-register--el1
+      ARM_CPU_FILE="/sys/devices/system/cpu/cpu0/regs/identification/midr_el1"
+
+      # https://gitlab.arm.com/telemetry-solution/telemetry-solution/-/blob/main/data/pmu/cpu/neoverse/neoverse-n1.json#L13
+      # N1:0xd0c; N2:0xd49; V1:0xd40;
+      Neoverse_N1="0xd0c"
+      Neoverse_N2="0xd49"
+      Neoverse_V1="0xd40"
+      hex_mask="0xfff0"
+      if [ -f "$ARM_CPU_FILE" ]; then
+        hex_ARM_CPU_DETECT=`cat $ARM_CPU_FILE`
+        ARM_CPU_PRODUCT=$(python -c "print(hex((int('$hex_ARM_CPU_DETECT', 16) & int('$hex_mask', 16)) >> 4))")
```

Comment:
python might cause portability issues since some platforms have python3 etc. We can use bash to do the arithmetic.
`printf "%d\n" $hex_ARM_CPU_DETECT`.

Reply:
Updated.
Thanks for your comments.